Replicator/tests/ReplicatorCollectionSGTest.cc

```diff
@@ -722,6 +722,103 @@ TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Resolve Conflict SG", "[.SyncServe
     }
 }
 
+TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Auto Purge Enabled - Revoke Access - SGColl", "[.SyncServerCollection]") {
+    _authHeader = "Basic c2d1c2VyOnBhc3N3b3Jk"_sl; // "sguser:password" base64 encoded
+    slice docID = "apera-doc1"_sl;
+
+    // Setup pull filter:
+    _pullFilter = [](C4CollectionSpec collectionSpec, C4String docID, C4String revID,
+                     C4RevisionFlags flags, FLDict flbody, void *context) {
+        if ((flags & kRevPurged) == kRevPurged) {
+            ((ReplicatorAPITest*)context)->_counter++;
+            Dict body(flbody);
+            CHECK(body.count() == 0);
+        }
+        return true;
+    };
+
+    // One-shot pull setup
+    constexpr size_t collectionCount = 1;
+    std::array<C4CollectionSpec, collectionCount> collectionSpecs = {
+            Roses
+    };
+    std::array<C4Collection*, collectionCount> collections =
+            collectionPreamble(collectionSpecs, "sguser", "password");
+    std::array<C4ReplicationCollection, collectionCount> replCollections;
+    replCollections[0] = C4ReplicationCollection {
+        collectionSpecs[0], kC4Disabled, kC4OneShot,
+        nullslice, nullptr, _pullFilter, this
+    };
+    C4ParamsSetter paramsSetter = [&replCollections,&collectionCount](C4ReplicatorParameters& c4Params) {
+        c4Params.collectionCount = collectionCount;
+        c4Params.collections = replCollections.data();
+    };
+
+    // Setup onDocsEnded:
+    _enableDocProgressNotifications = true;
+    _onDocsEnded = [](C4Replicator* repl,
+                      bool pushing,
+                      size_t numDocs,
+                      const C4DocumentEnded* docs[],
+                      void* context) {
+        for (size_t i = 0; i < numDocs; ++i) {
+            auto doc = docs[i];
+            if ((doc->flags & kRevPurged) == kRevPurged) {
+                ((ReplicatorAPITest*)context)->_docsEnded++;
+            }
+        }
+    };
+
+    auto collRoses = collections[0];
+
+    // Put doc in remote DB, in channels a and b
+    sendRemoteRequest("PUT", docID.asString(), R"({"channels":["a", "b"]})");
```

Comment:
This will put the doc to the default collection. Have you checked that doc actually landed on the CB server?

Reply:
Yeah, because currently SGW only supports single collection, PUT request will put the document in the named collection. (If you are using a named collection with SGW, default collection is not used)

Reply:
BUT when SGW supports multiple collections, that could change, so it would probably be a good idea for me to specify the collection to future-proof the test.

Reply:
You're right, I heard the same thing: they only more collection, default or named. I prefer to do test on the named-collection. I have extended sendRemoteRequest to include collection path in the arguments. I cannot send the PR because test is not passing. I have submitted a ticket against SGW.